zebra-chain/src/orchard/note/nullifiers.rs

```diff
@@ -46,9 +49,17 @@ impl Hash for Nullifier {
     }
 }
 
-impl From<[u8; 32]> for Nullifier {
-    fn from(bytes: [u8; 32]) -> Self {
-        Self(pallas::Base::from_bytes(&bytes).unwrap())
+impl TryFrom<[u8; 32]> for Nullifier {
+    type Error = SerializationError;
+
+    fn try_from(bytes: [u8; 32]) -> Result<Self, Self::Error> {
+        let possible_point = pallas::Base::from_bytes(&bytes);
+
+        if possible_point.is_some().into() {
+            Ok(Self(possible_point.unwrap()))
+        } else {
+            Err(SerializationError::Parse("Invalid pallas::Base value"))
```

Comment:
Since this message reveals implementation information, we should contextualize it:

```suggestion
            Err(SerializationError::Parse("Invalid pallas::Base value for orchard Nullifier"))
```

Reply:
Added at https://github.com/ZcashFoundation/zebra/pull/2267/commits/46edb646ca069c18d71659999014c3b3df58aaef
Thanks.